server/validate.go

```diff
@@ -106,47 +106,62 @@ func validateGetEntryAndProofRequest(req *trillian.GetEntryAndProofRequest) erro
 }
 
 func validateQueueLeafRequest(req *trillian.QueueLeafRequest) error {
-	if req.Leaf == nil {
-		return status.Error(codes.InvalidArgument, "QueueLeafRequest.Leaf empty")
-	}
-	if err := validateLogLeaf(req.Leaf); err != nil {
-		// validateLogLeaf errors are meant to chain nicely with "Leaf."
-		return status.Errorf(codes.InvalidArgument, "QueueLeafRequest.Leaf.%v", err)
-	}
-	return nil
+	return validateLogLeaf(req.Leaf, "QueueLeafRequest.Leaf")
+}
+
+func validateAddSequencedLeafRequest(req *trillian.AddSequencedLeafRequest) error {
+	return validateLogLeaf(req.Leaf, "AddSequencedLeafRequest.Leaf")
 }
 
 func validateQueueLeavesRequest(req *trillian.QueueLeavesRequest) error {
-	if len(req.Leaves) == 0 {
-		return status.Error(codes.InvalidArgument, "QueueLeavesRequest.Leaves empty")
+	return validateLogLeaves(req.Leaves, "QueueLeavesRequest")
```

Comment:
Can we not just remove this and the 2 functions above and call the validateLogXXX() functions directly?

Reply:
Done.